iroh-base/src/node_addr.rs

```diff
@@ -46,6 +46,18 @@ pub struct NodeAddr {
     pub info: AddrInfo,
 }
 
+impl PartialOrd for NodeAddr {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        self.node_id.partial_cmp(&other.node_id)
+    }
+}
+
+impl Ord for NodeAddr {
```

Comment:
I don't like this. This means that you can construct two NodeAddrs such that `a.cmp(b) == Ordering::Equal` yet `a != b`. I think this is forbidden or at least heavily discouraged. And if it is not - it should be...

Reply:
fixed